configuration-api/main/advancedAuth.js

```diff
@@ -27,11 +36,28 @@ const typeDefs = gql`
     language: String!
     lastTenantSelected: String
   }
+  type ResourceType {
+    name: String!
+    userID: String!
+    tenantID: String!
+  }
+  type ResourceEndpoint {
+    name: String!
+    resourceTypeName: String!
+    nameAndID: String!
+  }
   type Query {
     listTenants(tenantNames: [String]!): [TenantConfiguration]
     getUserPreferences(userName: String!): [UserPreferencies]
+    getUserResourceType(tenantID: String!): [ResourceType]
+    getEndpoints(resourceTypeName: String!): [ResourceEndpoint]
```

Comment:
if resource type are per tenants, it could be that the same resource type name is used by two (or more) different tenants.  this means that getEndpoints needs to take that into account.